### PR TITLE
Ground Station Logging

### PR DESCRIPTION
**Pull request by jonapap**

### CODE REVIEW

**Review by MarwanMashaly1 — APPROVED** [review]

File: libraries/messages/src/logging/macros.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+macro_rules! display_event {
+    ($( [$e:ident, $s:literal$(,)? $( $p:ident ),* ] ),*) => {
```
Comment (NoahSprenger) [inline libraries/messages/src/logging/macros.rs:2]: 😵‍💫 I am confused. Not very self documenting, maybe a brief comment of what is going on.

File: libraries/messages/src/lib.rs [hunk 2]
```diff
@@ -47,6 +50,7 @@ pub struct Message {
 pub enum Data {
```
Comment (NoahSprenger) [inline libraries/messages/src/lib.rs:50]: This is failing; "the trait bound `Log: TS` is not satisfied".

File: libraries/common-arm/src/error/hydra_error.rs [hunk 3]
```diff
@@ -68,7 +84,41 @@ impl<T, E> SpawnError for Result<T, E> {
     fn spawn_error(self, task: &'static str) -> Result<(), HydraError> {
         match self {
             Ok(_) => Ok(()),
-            Err(_) => Err(HydraError::SpawnError(task)),
+            // Err(_) => Err(HydraError::SpawnError(task)),
```
Comment (NoahSprenger) [inline libraries/common-arm/src/error/hydra_error.rs:87]: Can this line be removed?

**Review by NoahSprenger — CHANGES_REQUESTED** [review]

File: libraries/messages/src/lib.rs (hunk 2 above)
Comment (jonapap) [reply libraries/messages/src/lib.rs:50]: Since those logs are never sent to ground station, I've decided to remove the derive for that enum.

**Review by NoahSprenger — APPROVED** [review]
Awesome!!

**Pull request merged**